modules/reindex/src/main/java/org/elasticsearch/index/reindex/StartReindexJobAction.java

```diff
@@ -0,0 +1,184 @@
+//    public static final String NAME = "cluster:admin/reindex/start_reindex";
+//    public static final String NAME = "indices:data/reindex/start_reindex";
+
+    private StartReindexJobAction() {
+        super(NAME);
+    }
+
+    @Override
+    public StartReindexJobAction.Response newResponse() {
+        return new StartReindexJobAction.Response();
+    }
+
+    public static class Request extends AcknowledgedRequest<Request> implements ToXContentObject, CompositeIndicesRequest {
+
+        private ReindexRequest reindexRequest;
+        private boolean waitForCompletion = false;
+
+        public Request() {
+
+        }
+
+        public Request(ReindexRequest reindexRequest) {
+            this.reindexRequest = reindexRequest;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            reindexRequest = new ReindexRequest(in);
+            waitForCompletion = in.readBoolean();
+        }
+
+//        public static Request fromXContent(final XContentParser parser, final String id) throws IOException {
+//            Request request = new Request();
+//            request.setReindexJob(ReindexJob.fromXContent(parser));
+//            return request;
+//        }
+
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            super.readFrom(in);
+            reindexRequest = new ReindexRequest(in);
+            waitForCompletion = in.readBoolean();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            reindexRequest.writeTo(out);
+            out.writeBoolean(waitForCompletion);
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) {
+            return builder;
+        }
+
+        public ReindexRequest getReindexRequest() {
+            return reindexRequest;
+        }
+
+        public void setWaitForCompletion(boolean waitForCompletion) {
+            this.waitForCompletion = waitForCompletion;
+        }
+
+        public boolean getWaitForCompletion() {
+            return waitForCompletion;
+        }
+    }
+
+    public static class Response extends AcknowledgedResponse {
+
+        static final ParseField ACKNOWLEDGED = new ParseField("acknowledged");
+        static final ParseField TASK_ID = new ParseField("task_id");
+        static final ParseField REINDEX_RESPONSE = new ParseField("reindex_response");
+
+        private static final ConstructingObjectParser<Response, Void> PARSER = new ConstructingObjectParser<>(
+            "start_reindex_response", true, args -> new Response((boolean) args[0], (String) args[1], (BulkByScrollResponse) args[2]));
+
+        static {
+            PARSER.declareBoolean(ConstructingObjectParser.constructorArg(), ACKNOWLEDGED);
```

Comment:
this parser already exists in superclass. Reuse it and remove ACKNOWLEDGED in this class